shared/constants/transaction.js

```diff
@@ -249,6 +249,10 @@ export const TRANSACTION_GROUP_CATEGORIES = {
  *  ones, are added to the controller state in an unapproved status. When this
  *  happens we fire the Transaction Added event to show that the transaction
  *  has been added to the user's MetaMask.
+ * @property {'Transaction UI Action'} UI_ACTION - When an unapproved transaction
```

Comment:
These types of events should *modify properties of the other events* Like for example, if the user edits gas it should be modifying properties on the -added- fragment ... that fragment will ultimately result in the approved action when the user approves the tx...and then we can notate that the user modified the gas

Reply:
For some of the events it is possible to have modified properties in fragment but some events are like `user clicked edit link` or  `user open advance modal`. Which are just user action and no property update.

May be we keep only later ones in `UI_ACTION` action and others can have different action like `GAS_UPDATED` with property updates.

Reply:
I think, and could be wrong, that we'd prefer to have a 'user_clicked_edit_gas' property on the 'Transaction Approved' event -- such that we can see of the transactions approved by a user, what percentage attempted to have gas edited. Versus which percent actually modified the gas. @kevinghim could you tell me what you think? If we only track the click event separately it can tell us what percentage of *users* have clicked edit gas while transacting, but if we track it on the transaction event we can see what percentage of transactions users have attempted to edit gas on. Does that make sense?

Reply:
> we'd prefer to have a 'user_clicked_edit_gas' property on the 'Transaction Approved' event -- such that we can see of the transactions approved by a user, what percentage attempted to have gas edited.
This is correct. Ultimately, at the event of approval, we'd want to track not only the percentage of approval based on where it was approved (transaction screen, edit screen, advanced edit screen, swap), but also what was edited (gas option level, amount in preconfigured levels (L/M/H), or manual).

Reply:
@brad-decker : this is actually not clear to me, here are you suggesting that we capture these events in txMeta and add as property to transaction approved event ?